src/ui/public/vislib/visualizations/point_series/heatmap_chart.js

```diff
@@ -0,0 +1,268 @@
+              nextVal = Math.ceil(nextVal);
+            }
+            label = `${val} - ${nextVal}`;
+          }
+        }
+        labels.push(label);
+      }
+      return labels;
+    }
+
+    getHeatmapColors(cfg) {
+      const colorsNumber = cfg.get('colorsNumber');
+      const invertColors = cfg.get('invertColors');
+      const colorSchema = cfg.get('colorSchema');
+      const labels = this.getHeatmapLabels(cfg);
+      const colors = {};
+      for (const i in labels) {
+        if (labels[i]) {
+          const val = invertColors ? 1 - i / colorsNumber : i / colorsNumber;
+          colors[labels[i]] = getColor(val, colorSchema);
+        }
+      }
+      return colors;
+    }
+
+    addSquares(svg, data) {
+      const xScale = this.getCategoryAxis().getScale();
+      const yScale = this.handler.valueAxes[1].getScale();
+      const zScale = this.getValueAxis().getScale();
+      const ordered = this.handler.data.get('ordered');
+      const tooltip = this.baseChart.tooltip;
+      const isTooltip = this.handler.visConfig.get('tooltip.show');
+      const isHorizontal = this.getCategoryAxis().axisConfig.isHorizontal();
+      const colorsNumber = this.handler.visConfig.get('colorsNumber');
+      const setColorRange = this.handler.visConfig.get('setColorRange');
+      const colorsRange = this.handler.visConfig.get('colorsRange');
+      const color = this.handler.data.getColorFunc();
+      const labels = this.handler.visConfig.get('legend.labels');
+      const zAxisConfig = this.getValueAxis().axisConfig;
+      const zAxisFormatter = zAxisConfig.get('labels.axisFormatter');
+      const showLabels = zAxisConfig.get('labels.show');
+
+      const layer = svg.append('g')
+        .attr('class', 'series');
+
+      const squares = layer
+        .selectAll('g.square')
+        .data(data.values);
+
+      squares
+        .exit()
+        .remove();
+
+      let barWidth;
+      if (this.getCategoryAxis().axisConfig.isTimeDomain()) {
+        const { min, interval } = this.handler.data.get('ordered');
+        const start = min;
+        const end = moment(min).add(interval).valueOf();
+
+        barWidth = xScale(end) - xScale(start);
+        if (!isHorizontal) barWidth *= -1;
+      }
+
+      function x(d) {
+        return xScale(d.x);
+      }
+
+      function y(d) {
+        return yScale(d.series);
+      }
+
+      const [min, max] = zScale.domain();
+      function getColorBucket(d) {
+        let val = 0;
+        if (setColorRange && colorsRange.length) {
+          const bucket = _.find(colorsRange, range => {
+            return range.from <= d.y && range.to > d.y;
+          });
+          return bucket ? colorsRange.indexOf(bucket) : -1;
+        } else {
+          if (isNaN(min) || isNaN(max)) {
+            val = colorsNumber - 1;
+          } else {
+            val = (d.y - min) / (max - min); /* get val from 0 - 1 */
+            val = Math.floor(val * (colorsNumber - 1));
```

Comment:
I take that back about 748 being a bad example number, it should be in the last bucket.

0: 0-225
1: 225-450
2: 450-675
3: 675-900 <- 748 should be here